modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -549,4 +549,10 @@ private void setMessageProcessorState(ProcessorState state) {
 		registry.newNonEmptyResource(REG_PROCESSOR_BASE_PATH + getName(), false, "text/plain", state.toString(),
 				MP_STATE);
 	}
+
+    private void deleteMessageProcessorState() {
+        if (registry.getResourceProperties(REG_PROCESSOR_BASE_PATH + getName()) != null) {
+            registry.delete(REG_PROCESSOR_BASE_PATH + getName());
```

Comment:
We have the same check inside WSO2Registry.delete() as well. Canwe remove that since it's handled from a higher level?